### ISSUE TITLE
We should define the behavior of ‘use’ in terms of Web Components

### ISSUE THREAD

**Issue by nikosandronikos** [w3c/svgwg#99]
See https://svgwg.org/svg2-draft/struct.html#issue28
We should define the behavior of ‘use’ in terms of Web Components. See ACTION-3729 (Tab)

http://www.w3.org/Graphics/SVG/WG/track/actions/3729

**Comment by AmeliaBR** [comment]
We won't be able to define in terms of "web components" per se, but should be able to define in terms of the shadow DOM spec, and therefore make it compatible with Shadow DOM event handling and CSS shadow dom selectors.

Will try to get a good text for the next telcon post-F2F London.

**Comment by shepazu** [comment]
We need to reconsider during CR whether to include a normative reference to Shadow DOM, which is still an unstable WD, or to revert to previous wording and defer Shadow DOM to SVG 3.

**Comment by AmeliaBR** [comment]
@shepazu 

I don't think we want to revert the changes. That would be very unfortunate for long-term goals of improved interoperability.

If Shadow DOM spec doesn't stabilize, we may need to introduce a few additional details so that the use-element shadow DOM can be implemented without a normative reference to it.  Then we could have some informative paragraphs about how the two specs are intended.

PS, You'll need to confirm with the Shadow DOM editors, but the intention may be to simply integrate it in core DOM, instead of having a separate spec.  That is what's been done on WHATWG side.  In fact, we could probably make it a normative reference to the WHATWG DOM standard instead of to W3C Shadow DOM.  The only reason I didn't do that was because we have normative references to W3C DOM 4 elsewhere.

**Comment by shepazu** [comment]
> I don't think we want to revert the changes. That would be very unfortunate for long-term goals of improved interoperability.

It wouldn't change anything for better or worse. It would simply remain the same as SVG1.1, which would mean that implementers wouldn't need to make changes that might need to change again to match an updated Shadow DOM and future versions of SVG. This would be the most conservative option.

I wouldn't mind informative text, but we'd need to consider what the goal there is.

> PS, You'll need to confirm with the Shadow DOM editors, but the intention may be to simply integrate it in core DOM, instead of having a separate spec.

It's not a matter of where it's defined, but rather how stable it is.
